inc/related-content.php

```diff
@@ -758,8 +758,11 @@ protected function add_from_query( $q, $reversed = FALSE ) {
 		while ( $q->have_posts() ) {
 			$q->the_post();
 			// add this post if it's new
-			if ( ! in_array( $q->post->ID, $this->post_ids ) ) {	// only add it if it wasn't already there
-				$this->post_ids[] = (int) trim( $q->post->ID );
+			if (
+				! in_array( $q->post->ID, $this->post_ids ) 	// only add it if it wasn't already there
+				&& $q->post->ID != $this->post_id // do not add the id of the current post, because we do not want to ever return that
```

Comment:
I think we should allow type juggling here, because of the possibility that we may be comparing a string post ID to a numeric post ID.